Makefile

```diff
@@ -32,7 +32,8 @@ server:
 clean:
 	yarn run clean
 
+# test_unit_codecov runs unit tests with code coverage turned on and
+# submits the coverage to codecov.io
 .PHONY: test_unit_codecov
-test_unit_codecov: test_unit
-	npm install codecov -g
-	codecov -f .coverage/coverage-final.json
+test_unit_codecov:
+	yarn run test:coverage && yarn run codecov
```

Comment:
does yarn run automatically install codecov?

Reply:
ohh nevermind you added it as a dependency. But why?
Its not really required to install or execute the code. IMO this is not ideal

Reply:
Having it as a global dependency (even in travis) is also not ideal. This way, the version we are using is stored in our lockfile and will remain consistent unless we specifically upgrade. It's also stored as a dev dependency, not a runtime dependency. 
I can move out of our dependencies and just install it during the travis builds. But either way, it should probably be installed with yarn, since that's what we use for package management.

Reply:
@kumare3 I updated it to install codecov only on Travis as part of the `install` step. It's important to do it here because Travis caches the result of the `install` step to make future runs faster. I think that's the middle ground between what I was trying to do and your suggestion (keeps it out of our dependencies for installing/running, keeps it out of global package scope, allows it to be cached by Travis).

Reply:
ya sorry I am not really aware of the tooling, yesterday was the first time I was playing around with all the react and node tooling. I learnt a little bit - jest, npm, yarn etc